RELEASING.md

```diff
@@ -1,34 +1,74 @@
+
+## Engine
 
 ### Prep
 
+- Ensure you have GHCR credentials for your Dagger Github account configured ([instructions](https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-container-registry#authenticating-to-the-container-registry)).
+- Ensure [Mage is installed](https://magefile.org/).
+- Ensure that CI is green for all tests in main
+
 ```console
 git pull
 git status  # make sure everything is clean
 ```
 
-### Release Engine
+### Bump version tags
+
+- Modify `EngineImageRef` in `internal/mage/engine.go` to use a tag set to the engine version (e.g. `v0.3.0`).
 
 ```console
 export ENGINE_VERSION=v0.3.0
 git tag $ENGINE_VERSION
 git push origin $ENGINE_VERSION
+mage engine:publish
 ```
 
-### Bump Engine dependency
+## Go SDK
+
+### Prep
+
+```console
+git pull
+git status  # make sure everything is clean
+```
+
+### Bumping an Engine dependency
+
+The following is needed if the SDK requires a new version of the engine.
 
 ```console
-export ENGINE_VERSION=v0.3.0
 cd ./sdk/go
-go get github.com/dagger/dagger@${ENGINE_VERSION}
-go mod tidy
 git checkout -b go-sdk-update
-git commit go.mod go.sum -m 'sdk: go: bump engine dependency'
+```
+
+1. Obtain the full image ref, including SHA, from our GHCR repo for the version of the engine the SDK should depend on.
+
+- Our GHCR repo can be found [here](https://github.com/orgs/dagger/packages/container/package/engine).
+- Locate the tag for the engine the SDK should depend on and note the digest of it.
+  - The digest will be prefixed with `sha256:` and look like `a10730b1d305644b87c21630d27d9e15d424014ee7ac9bd44f4504fcfe92a8ca`
+
+1. Update the value of `engineImageRef` in `client.go` to be `ghcr.io/dagger/engine:<version tag>@sha256:<digest>` ([line currently here](https://github.com/dagger/dagger/blob/349f8a1661f1a8422a8bef919a345bd6dd3f3d85/sdk/go/client.go#L19))
+
+```console
+git commit client.go -m 'sdk: go: bump engine dependency'
 gh pr create
 ```
 
+1. Ensure all tests in CI pass.
+1. Additionally, manually test provisioning by running the following commands on both Linux and MacOS hosts directly
+
+```console
+export SHORT_DIGEST=a10730b1d305644b # replace the value with the first 16 chars of the digest of the engine image used above
+docker rm -fv dagger-engine-$SHORT_DIGEST # ensure there's no existing image so we use the full provisioning code paths
+rm -rf ~/.cache/dagger/* # ensure there's no existing local engine-session binaries for same reason as above
+cd ../.. # go to the root of the dagger repo
+go test -v -count=1 $(pwd)/core/integration # run the engine tests on your host, which exercises the provisioning code paths of the Go SDK
```

Comment:
This should be fairly easy to automate as a mage step (the reason it isn't already is because we run tests dagger-in-dagger, which is great but skips the provisioning code paths)

Reply:
Though automating it on MacOS is not going to happen before the next release, so that will still be manual